src/main/java/ua/lviv/iot/parkingServer/logic/VehicleService.kt

```diff
@@ -0,0 +1,50 @@
+package ua.lviv.iot.parkingServer.logic
+
+import org.springframework.stereotype.Service
+import ua.lviv.iot.parkingServer.datastorage.VehicleFileStore
+import ua.lviv.iot.parkingServer.model.Vehicle
+import javax.annotation.PostConstruct
+import javax.annotation.PreDestroy
+
+@Service
+class VehicleService(private val vehicleFileStore: VehicleFileStore) {
+    private val vehicles: MutableMap<Long, Vehicle> = HashMap()
+    private var index: Long = 0L
+
+    fun findAllVehicles(): List<Vehicle> = ArrayList<Vehicle>(vehicles.values)
+
+    fun findVehicleById(id: Long): Vehicle? = vehicles[id]
+
+    fun addVehicle(vehicle: Vehicle): Vehicle {
+        index += 1
+        vehicle.vehicleId = index
+        vehicles[index] = vehicle
+        return vehicle
+    }
+
+    fun updateVehicle(id: Long, vehicle: Vehicle): Vehicle {
+        vehicle.vehicleId = id
+        vehicles[id] = vehicle
+        return vehicle
+    }
+
+    fun deleteVehicle(id: Long): Vehicle? = vehicles.remove(id)
+
+    @PreDestroy
+    fun saveVehicleData() {
+        val vehicleList: List<Vehicle> = vehicles.values.stream().toList()
+        vehicleFileStore.saveRecords(vehicleList)
+    }
+
+    @PostConstruct
+    fun vehicleDataToHashmap() {
+        val vehicleList: List<Vehicle> = vehicleFileStore.readRecords()
+        for (vehicle: Vehicle in vehicleList) {
+            index += 1
+            if (vehicle.vehicleId > index) {
+                index = vehicle.vehicleId
+            }
+            vehicles[vehicle.vehicleId] = vehicle
+        }
```

Comment:
This logic is overcomplicated.

Suggestion:
```kotlin
  val idToVehicle: Map<Long, Vehicle> =  vehicleFileStore.readRecords().associateBy { it.vehicleId }
  vehicles.putAll(idToVehicle)
  
  val maxId = idToVehicle.max() ?: 0
  index = max(maxId, idToVehicle.size())
```

See https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/associate-by.html

Reply:
✅